### PR TITLE
Enable composeModelMatrix for meshlayers

### PR DESCRIPTION
**Pull request by xintongxia**
#### Background
For v8 API audit

#### Change List
- compose modelMatrix with instance modelMatrix for mesh layers
- add tests
- update documentations
- update mesh example app

### CODE REVIEW

File: modules/mesh-layers/src/utils/matrix.js [hunk 1]
```diff
@@ -153,3 +153,22 @@ export const MATRIX_ATTRIBUTES = {
...
+      ? COORDINATE_SYSTEM.LNGLAT
+      : COORDINATE_SYSTEM.CARTESIAN;
+  }
+
+  let composeModelMatrix = false;
+  if (
+    coordinateSystem === COORDINATE_SYSTEM.CARTESIAN ||
+    coordinateSystem === COORDINATE_SYSTEM.METER_OFFSETS
+  ) {
+    composeModelMatrix = true;
```
Comment (Pessimistress) [inline modules/mesh-layers/src/utils/matrix.js]: Just `return true`

File: package.json [hunk 2]
```diff
@@ -44,9 +44,9 @@
     "@luma.gl/engine": "^8.0.0-alpha.12",
     "@luma.gl/test-utils": "^8.0.0-alpha.10",
     "@probe.gl/bench": "^3.2.0-beta.3",
-    "@probe.gl/test-utils": "^3.2.0-beta.3",
-    "@loaders.gl/csv": "^1.3.3",
-    "@loaders.gl/polyfills": "^1.3.3",
+    "@probe.gl/test-utils": "^3.1.1",
```
Comment (Pessimistress) [inline package.json]: Why are we downgrading this?

File: examples/website/mesh/app.js [hunk 3]
```diff
@@ -96,6 +129,7 @@ export default class App extends PureComponent {
         controller={true}
         layers={layers}
         effects={[lightingEffect]}
+        onAfterRender={this._rotate}
```
Comment (Pessimistress) [inline examples/website/mesh/app.js]: Why is this needed if rotation is already on an animation loop?

File: test/render/test-cases.js [hunk 4]
```diff
@@ -1747,5 +1771,85 @@ export const TEST_CASES = [
...
+        modelMatrix: new Matrix4().rotateX((-45 / 180) * Math.PI),
+        getPosition: d => d.position,
+        getColor: d => d.color,
+        getOrientation: d => d.orientation
+      })
+    ],
+    goldenImage: './test/render/golden-images/simple-mesh-layer-cartesian.png'
+  },
+  {
+    name: 'simple-mesh-layer-meter-offsets',
```
Comment (Pessimistress) [inline test/render/test-cases.js]: I think it's more important to test the LNGLAT use case

File: test/render/test-cases.js [hunk 5]
```diff
@@ -18,14 +18,34 @@ import {
...
 import {Fp64Extension, PathStyleExtension} from '@deck.gl/extensions';
+import {SimpleMeshLayer} from '@deck.gl/mesh-layers';
+import {Matrix4} from 'math.gl';
 
 const effect1 = new PostProcessEffect(noise);
 const effect2 = new PostProcessEffect(vignette);
 
 const ICON_ATLAS = './test/render/icon-atlas.png';
 
+const SAMPLE_DATA = (([xCount, yCount], spacing) => {
```
Comment (Pessimistress) [inline test/render/test-cases.js]: Can we move this to layer-browser's data samples? This file is getting too big.

File: modules/mesh-layers/src/utils/matrix.js [hunk 6]
```diff
@@ -153,3 +153,17 @@ export const MATRIX_ATTRIBUTES = {
...
+  // only apply composeModelMatrix when in cartesian or meter_offsets coordinate system
+  if (coordinateSystem === COORDINATE_SYSTEM.DEFAULT) {
+    coordinateSystem = viewport.isGeospatial
+      ? COORDINATE_SYSTEM.LNGLAT
+      : COORDINATE_SYSTEM.CARTESIAN;
+  }
+
+  return (
+    coordinateSystem === COORDINATE_SYSTEM.CARTESIAN ||
+    coordinateSystem === COORDINATE_SYSTEM.METER_OFFSETS
```
Comment (Pessimistress) [inline modules/mesh-layers/src/utils/matrix.js]:
```js
|| (coordinateSystem === COORDINATE_SYSTEM.DEFAULT && !viewport.isGeospatial)
```

File: modules/mesh-layers/src/utils/matrix.js [hunk 7]
```diff
@@ -153,3 +153,17 @@ export const MATRIX_ATTRIBUTES = {
     }
   }
 };
+
+export function shouldComposeModelMatrix(viewport, coordinateSystem) {
```
Comment (Pessimistress) [inline modules/mesh-layers/src/utils/matrix.js]: Add comments here to document why the model matrix is only composed for these coordinate systems

File: test/render/test-cases.js (hunk 4 above)
Comment (xintongxia) [reply test/render/test-cases.js]: I have one test for `COORDINATES.DEFAULT`, which is lnglat

File: test/render/test-cases.js [hunk 8]
```diff
@@ -1734,5 +1744,85 @@ export const TEST_CASES = [
...
+      pitch: 0,
+      bearing: 0
+    },
+    layers: [
+      new SimpleMeshLayer({
+        id: 'simple-mesh-layer-lnglat',
+        data: dataSamples.points.slice(0, 10),
+        mesh: cube,
+        sizeScale: 100,
+        modelMatrix: new Matrix4().rotateX((-45 / 180) * Math.PI),
```
Comment (Pessimistress) [inline test/render/test-cases.js]: LNGLAT coordinates should not be rotated
